pkg/errors/errors.go

```diff
@@ -0,0 +1,18 @@
+package errors
+
+import (
+	"github.com/directxman12/k8s-prometheus-adapter/pkg/metrics"
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/provider"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func NewMetricNotFoundError(resource schema.GroupResource, metricName string) error {
+	metrics.Errors.WithLabelValues("not_found").Inc()
```

Comment:
I see what this code is trying to achieve, namely to reduce reduce the repetition of metric updates in the caller code, but this construct is not idiomatic. This method should do what it implies, creating a new error and not introduce side effects.

Suggestion: let the `MetricNotFoundError` be a sentinal error and react on that in the calling code.

Reply:
The caller in most cases is somewhere inside `apiserver` package. I've got no qualms about not using utility functions for this, but it _is_ going to be a lot less DRY. I'll update the PR to reflect that, and you can decide if you prefer it to this option.

Reply:
I've updated the PR to not introduce an `errors` package, preserving the previous behavior. I didn't see any opportunity to treat the error as a sentinel at a higher level, in a way that would improve clarify or reduce repetition.

(e.g., several of the functions are called in multiple places, so it would be more code & more complicated code farther from the actual causal location to check for certain types errors & report metrics about them.)

The places where we really _could_ reduce repetition are inside apiserver, not in this code base.

(Additionally, since the metrics are no longer global, each call site is even slightly more verbose to handle the case where, e.g. during testing, metrics have not been instantiated.)

Happy to adjust for any further feedback you might have.